sdks/java/core/src/test/java/org/apache/beam/sdk/io/FileSystemsTest.java

```diff
@@ -186,6 +189,56 @@ public void testRenameIgnoreMissingFiles() throws Exception {
         containsInAnyOrder("content3"));
   }
 
+  @Test
+  public void testRenameOverwriteExistingFilesThrows() throws Exception {
+    // Simulate a filesystem which will not overwrite existing files
+    FileSystem<LocalResourceId> fs = new LocalFileSystem(new CopyOption[0]);
+    thrown.expect(FileAlreadyExistsException.class);
+    executeRename(fs, false);
+  }
+
+  @Test
+  public void testRenameOverwriteExistingFiles() throws Exception {
+    // Simulate a filesystem which will not overwrite existing files
```

Comment:
which will ovewrite

Reply:
Please see above. We're testing FileSystems behavior when a FileSystem implementation does not behave like a LocalFileSystem.

Reply:
Sorry to chime in. I think what this test intends to do is to force the behavior that happens in other filesystems (HDFS). I don't see how this could be done otherwise.

Reply:
Thanks for the clarification.